services/data_plane_manager/src/main/java/com/futurewei/alcor/dataplane/service/impl/RouterService.java

```diff
@@ -63,17 +61,22 @@ public void buildRouterState(InternalRouterInfo routerInfo, InternalSubnetRoutin
         for (InternalRoutingRule routingRule: routingRules) {
             Router.RouterConfiguration.RoutingRule.Builder routingRuleBuilder = Router.RouterConfiguration.RoutingRule.newBuilder();
             routingRuleBuilder.setOperationType(getOperationType(routingRule.getOperationType()));
+
             routingRuleBuilder.setId(routingRule.getId());
-            routingRuleBuilder.setName(routingRule.getName());
+            if (routingRule.getName() != null)
+            {
+                routingRuleBuilder.setName(routingRule.getName());
+            }
             routingRuleBuilder.setDestination(routingRule.getDestination());
             routingRuleBuilder.setNextHopIp(routingRule.getNextHopIp());
             routingRuleBuilder.setPriority(routingRule.getPriority());
 
+
             if (routingRule.getRoutingRuleExtraInfo() != null) {
                 Router.RouterConfiguration.RoutingRuleExtraInfo.Builder extraInfoBuilder = Router.RouterConfiguration.RoutingRuleExtraInfo.newBuilder();
                 extraInfoBuilder.setDestinationType(getDestinationType(
                         routingRule.getRoutingRuleExtraInfo().getDestinationType()));
-                extraInfoBuilder.setNextHopMac(routingRule.getRoutingRuleExtraInfo().getNextHopMac());
+
```

Comment:
@DavidLiu506 Why you removed `extraInfoBuilder.setNextHopMac(routingRule.getRoutingRuleExtraInfo().getNextHopMac());`